channel-connector-starter/src/main/java/com/salesforce/einsteinbot/connector/framework/EinsteinBotAutoConfiguration.java

```diff
@@ -32,27 +33,43 @@ public class EinsteinBotAutoConfiguration {
   @Bean
   @ConditionalOnMissingBean
   public Cache getCache() {
-    EinsteinBotConfiguration.Cache cache = einsteinBotConfiguration.getCache();
-    if (cache.getRedisUrl() != null) {
-      return new RedisCache(cache.getTtlSeconds(), cache.getRedisUrl());
+    return createCacheFromCacheConfig(einsteinBotConfiguration.getCache());
+  }
+
+  private Optional<Cache> getCacheForOAuth(EinsteinBotConfiguration.Cache cacheConfig) {
+    if (cacheConfig != null && cacheConfig.getTtlSeconds() != null){
```

Comment:
This will still get cache TTL for session cache. It is not using `sfdc.einstein.bots.oauth.cache.ttlseconds`

Reply:
We are using the same type `EinsteinBotConfiguration.Cache` for both top-level Cache and one with `oauth`. But correct cache configuration object `getCacheForOAuth(oauthConfig.getCache()` is passed to this method. 
I also updated the unit test to verify this.
Here is the actual code snippet:

```
EinsteinBotConfiguration.OAuth oauthConfig = einsteinBotConfiguration.getoAuth();
    return JwtBearerOAuth.with()
        .privateKeyFilePath(oauthConfig.getPrivateKeyFile())
        .loginEndpoint(oauthConfig.getLoginEndpoint())
        .connectedAppId(oauthConfig.getConnectedAppId())
        .connectedAppSecret(oauthConfig.getConnectedAppSecret())
        .userId(oauthConfig.getUserId())
        .cache(getCacheForOAuth(oauthConfig.getCache()))
        .build();
```

Reply:
oh! thanks, got it.